Update _stats_py.py

Mitraansh opened the pull request:
Add a comment in the zscore definition that we have to use zmap for the standardization as if we go directly for score there is a difference between the value that we are getting from zscore=(x-mean)/std.

<!-- 
Thanks for contributing a pull request! Please ensure that
your PR satisfies the checklist before submitting:
http://scipy.github.io/devdocs/dev/contributor/development_workflow.html#checklist-before-submitting-a-pr

Also, please name and describe your PR as you would write a
commit message:
http://scipy.github.io/devdocs/dev/contributor/development_workflow.html#writing-the-commit-message

Note that we are a team of volunteers; we appreciate your
patience during the review process.

Again, thanks for contributing!
-->

#### Reference issue
<!--Example: Closes gh-WXYZ.-->
This issue is for the values of zscore. If we use the mathematical formula that is z=(x-mean)/std then sometimes we get a different value when we use zscore function

#### What does this implement/fix?
<!--Please explain your changes.-->
There is no a such exact solution but we can notify users why this issue may be there and so we can use zmap separately and then go for zscore

#### Additional information
<!--Any additional information you think is important.-->
Or what else can be done is by notifying the user that there should not be null values and we make a function return (x-mean)/std directly.

Mitraansh commented:
I have added only the comment but have not updated the code function.

mdhaber commented:
> If we use the mathematical formula that is z=(x-mean)/std then sometimes we get a different value when we use zscore function

Can you give an example? (Not in the code but in a comment below.)

Mitraansh commented:
what I am trying to convey is if we have an original dataset which has nan values and we don't want to change the original dataset so we have another dataset which is a copy of the original and we updated the nan values in that using a certain techniques. If a person directly applies zscore on the original data values will be different as zscore is not updating the nan values by replacement techniques like mean, median or some other technique so the values will be different from (x-mean)/std so if they check the reason in the code there is no such definition for that situation. Now if a person doesn't want to update the real set he can get zscore by zmap(original, updated)

Mitraansh commented:
There should be an explanation regarding that for the beginners who doesn't understand the zscore calculation that well or the zscore function should update the nan values by mean, median or mode or other techniques.

Mitraansh commented:
![Screenshot 2023-02-26 214852](https://user-images.githubusercontent.com/59146380/221422805-809ea3fa-e7ca-4b74-995e-f75fc8933e1a.png)
![Screenshot 2023-02-26 214930](https://user-images.githubusercontent.com/59146380/221422809-a210da2e-5332-41e9-9a86-b48f533ef549.png)
![Screenshot 2023-02-26 215005](https://user-images.githubusercontent.com/59146380/221422812-89b23c90-88e7-418a-9c3f-c223737cb718.png)
![Screenshot 2023-02-26 215018](https://user-images.githubusercontent.com/59146380/221422813-3cf41997-5fde-4425-94cd-bed80e4f19d4.png)
![Screenshot 2023-02-26 215042](https://user-images.githubusercontent.com/59146380/221422816-72befbec-0640-49ce-957f-9642b8937913.png)
![Screenshot 2023-02-26 215058](https://user-images.githubusercontent.com/59146380/221422817-86341c1f-7254-4634-be9d-c1f3d2355d38.png)

Mitraansh commented:
newdf is the old data frame and newdf1 is updated.

mdhaber commented:
@Mitraansh please provide a [minimal reproducible example](https://en.wikipedia.org/wiki/Minimal_reproducible_example): simple code we can run to see the issue you're describing. 

For example, suppose I wanted to report the issue that:

> BUG: stats.tsem: ignores `axis` argument

I would provide the minimum reproducible example:

```python3
import numpy as np
from scipy import stats
rng = np.random.default_rng(21165541256537398)
a = rng.random(size=(4, 5))
stats.tsem(a)  # 0.06862956713881908
stats.tsem(a, axis=0)  # 0.06862956713881908, but should be 1d array of size 5
stats.tsem(a, axis=1)  # 0.06862956713881908, but should be 1d array of size 4
```
and note that:

> [stats.tsem](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.tsem.html) accepts argument `axis` but ignores it, always behaving as though `axis=None`.

melissawm commented on scipy/stats/_stats_py.py:
```diff
@@ -3120,6 +3120,8 @@ def gzscore(a, *, axis=0, ddof=0, nan_policy='propagate'):
     >>> ax.hist(gzscore(x), 50)
     >>> plt.show()
+    
+    Also for the zscore nomalize value you need to use zmap to get the correct zscore values
```
I agree this sentence is not very clear. Perhaps add a tiny reproducible example to show what you mean after writing something like

```suggestion
    To obtain the normalized zscore values, you can use the `zmap` function.
```
